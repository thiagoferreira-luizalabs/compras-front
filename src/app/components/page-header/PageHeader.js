import React from 'react';
import { Menu, Dropdown, Image, Icon } from 'semantic-ui-react'
import { Link } from "react-router-dom"
import Strips from "../strips/Strips"
import Logo from "../../../assets/img/magazineluiza.png"
import "./page-header.css"


export const ItemsMenu = (menus) => (
  Object.keys(menus).map((key, idx) => (
    <Dropdown key={idx} item text={menus[key].title} className={idx === 0 ? "active" : ""}>
      <Dropdown.Menu>
        {menus[key].subItems.map((item, idx) =>
          (<Dropdown.Item as={Link} key={idx} to={item.route}>{item.menuName}</Dropdown.Item>))}
      </Dropdown.Menu>
    </Dropdown>
  ))
)

const PageHeader = ({ onToggleSideBar, menus }) => (
  <div>
    <Strips />

    <Menu className="topbar-mobile" >
      <Menu.Item onClick={onToggleSideBar}><Icon name="bars" /></Menu.Item>
      <Menu.Item as={Link} to="/home" className="logo"><Image src={Logo} /></Menu.Item>
    </Menu>

    <Menu className="navbar">
      <Menu.Item as={Link} to="/">
        <Image src={Logo} className="logo" />
      </Menu.Item>

      {ItemsMenu(menus)}

      <Menu.Menu position='right' className="logout">
        <Dropdown item text='OlÃ¡, CBT_SILVA'>
          <Dropdown.Menu>
            <Dropdown.Item as={Link} to="/logout">Logout</Dropdown.Item>
          </Dropdown.Menu>
        </Dropdown>
      </Menu.Menu>

    </Menu>
  </div >
)

export default PageHeader