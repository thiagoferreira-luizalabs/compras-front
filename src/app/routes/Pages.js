import Theme from "../containers/theme/Theme"
import Home from "../containers/home/Home"


export const Pages = {
  Home: {
    name: "Home",
    menuName: "Inicio",
    route: "/home",
    component: Home
  },
  Theme: {
    name: "Theme",
    menuName: "Tema",
    route: "/theme",
    component: Theme
  }
}

export const Menus = {

  Comercial: {
    title: "Comercial",
    subItems: [
      Pages.Home,
      Pages.Home,
      Pages.Home,
    ]
  },
  Logistica: {
    title: "Log√≠stica",
    subItems: [
      Pages.Home,
      Pages.Home,
      Pages.Home,
    ]
  },
  Theme: {
    title: "Tema",
    subItems: [
      Pages.Theme
    ]
  },
}