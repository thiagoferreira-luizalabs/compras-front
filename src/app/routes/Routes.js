import React, { Component } from "react"
import { BrowserRouter as Router, Switch, Redirect } from "react-router-dom";
import AppRoute from "./AppRoute"
import App from "../App"
import AuthService from "../services/AuthService"
// Pages
import Login from "../containers/login/Login"
import Theme from "../containers/theme/Theme"
import Home from "../containers/home/Home"

class Routes extends Component {

  constructor(props) {
    super(props)
    this.state = { auth: AuthService.getAuth() }
    this.setAuth = this.setAuth.bind(this)
    this.logout = this.logout.bind(this)
  }

  setAuth() {
    AuthService.setAuth()
    this.setState({ auth: AuthService.getAuth() })
  }

  logout() {
    AuthService.logout()
    this.setState({ auth: AuthService.getAuth() })
    return <Redirect to={{ pathname: '/' }} />
  }

  render() {
    return (
      <Router>
        <Switch>
          <App auth={this.state.auth}>
            {/* Public Routes */}
            <AppRoute public exact {...this.state} path="/" component={Login}
              componentProps={{ setAuth: this.setAuth }} />

            {/* Private Routes */}
            <AppRoute private exact {...this.state} path="/home" component={Home} />
            <AppRoute private exact {...this.state} path="/theme" component={Theme} />

            <AppRoute exact {...this.state} path="/logout" component={() => this.logout()} />

          </App>
        </Switch>
      </Router>
    )
  }

}

export default Routes