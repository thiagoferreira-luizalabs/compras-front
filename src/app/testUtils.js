import React from "react"
import { configure as enzymeConfigure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import sinon from "sinon";
import { MemoryRouter } from "react-router-dom";

enzymeConfigure({ adapter: new Adapter() });

const asyncAssert = (callback, done) => {
  setTimeout(() => {
    try {
      callback()
      done()
    } catch (e) {
      done.fail(e)
    }
  }, 5)
}

const createFileChangeEvent = (file) => (
  {
    target: {
      files: [file]
    }
  }
)

const MemoryRouterHOC = (children) => (<MemoryRouter>{children}</MemoryRouter>)

const toJsonSnapshot = (wrapper) => toJson(wrapper, { mode: "deep", noKey: true })

export {
  shallow, mount, sinon, MemoryRouter, toJsonSnapshot, asyncAssert, createFileChangeEvent, MemoryRouterHOC
}
