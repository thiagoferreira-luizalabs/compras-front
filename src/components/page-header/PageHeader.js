import React from 'react';
import { Menu, Button } from 'semantic-ui-react'
import { Link } from "react-router-dom"

export default () => (
  <Menu>
    <Menu.Item header>Compras</Menu.Item>
    <Menu.Item name='aboutUs' />
    <Menu.Item name='jobs' />
    <Menu.Item name='locations' />
    <Menu.Item>
      <Button primary>Sign up</Button>
    </Menu.Item>

    <Menu.Item>
      <Link to="/logout"><Button>Logout</Button></Link> :
    </Menu.Item>
  </Menu>
)