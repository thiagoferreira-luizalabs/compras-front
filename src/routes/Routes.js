import React, { Component } from "react"
import { BrowserRouter as Router, Switch, Redirect } from "react-router-dom";
import AppRoute from "./AppRoute"
import App from "../App"
import AuthService from "../services/AuthService"
// Pages
import Login from "../containers/login/Login"
const Home = () => (<h1>Welcome to the homepage. You are logged in! Go corintia. </h1>)

class Routes extends Component {

  constructor(props) {
    super(props)
    this.state = { auth: AuthService.getAuth() }
    this.setAuth = this.setAuth.bind(this)
    this.logout = this.logout.bind(this)
  }

  setAuth() {
    AuthService.setAuth()
    this.setState({ auth: AuthService.getAuth() })
  }

  logout() {
    AuthService.logout()
    this.setState({ auth: AuthService.getAuth() })
    return <Redirect to={{ pathname: '/' }} />
  }

  render() {
    return (
      <Router>
        <Switch>
          <App auth={this.state.auth}>
            {/* Public Routes */}
            <AppRoute public exact {...this.state} path="/" component={Login}
              componentProps={{ setAuth: this.setAuth }} />

            {/* Private Routes */}
            <AppRoute private exact {...this.state} path="/home" component={Home} />

            <AppRoute exact {...this.state} path="/logout" component={() => this.logout()} />

          </App>
        </Switch>
      </Router>
    )
  }

}

export default Routes