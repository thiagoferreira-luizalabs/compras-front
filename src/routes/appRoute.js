import React from "react"
import { Route, Redirect } from "react-router-dom";

const AppRoute = ({ component: Component, ...routeProps, state }) => (
  <Route {...routeProps}
    render={props => {
      const auth = routeProps.isAuthenticated;

      //PUBLIC ROUTES
      if (routeProps.public) {
        return !auth ?
          <Component {...props} state={state} /> :
          <Redirect to={{ pathname: "/home" }} />;

        //PRIVATE ROUTES
      } else if (routeProps.private) {
        return auth ?
          <Component {...props} state={state} /> :
          <Redirect to={{ pathname: "/login", state: { from: props.location } }} />;

        // DEFAULT - REGULAR REACT-ROUTER OBJCET
      } else {
        return (<Component {...props} />)
      }

    }
    }
  />
);

export default AppRoute;