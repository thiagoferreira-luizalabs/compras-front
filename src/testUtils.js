import { configure as enzymeConfigure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import sinon from "sinon";
// import { MemoryRouter } from "react-router-dom";

enzymeConfigure({ adapter: new Adapter() });

const asyncAssert = (callback, done) => {
  setTimeout(() => {
    try {
      callback()
      done()
    } catch (e) {
      done.fail(e)
    }
  }, 5)
}

const createFileChangeEvent = (file) => (
  {
    target: {
      files: [file]
    }
  }
)

export {
  shallow, mount, sinon,  /*MemoryRouter,*/ toJson, asyncAssert, createFileChangeEvent
}
