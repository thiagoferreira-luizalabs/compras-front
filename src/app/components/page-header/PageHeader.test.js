import React from 'react';
import PageHeader from './PageHeader';
import { mount, toJsonSnapshot, MemoryRouterHOC } from "../../testUtils"
import { Menus } from "../../routes/Pages"

it("Renders page header", () => {
  let wrapper = mount(
    MemoryRouterHOC(<PageHeader />)
  )
  expect(toJsonSnapshot(wrapper)).toMatchSnapshot()
})