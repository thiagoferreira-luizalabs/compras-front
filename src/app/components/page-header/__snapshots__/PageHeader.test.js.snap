// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders page header 1`] = `
<div>
  <div>
    <div
      className="strips"
    >
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
    </div>
    <div
      className="clear"
    />
  </div>
  <div
    className="ui topbar-mobile menu"
  >
    <div
      className="item"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="bars icon"
      />
    </div>
    <a
      className="item logo"
      href="/home"
      onClick={[Function]}
    >
      <img
        className="ui image"
        src="magazineluiza.png"
      />
    </a>
  </div>
  <div
    className="ui navbar menu"
  >
    <a
      className="item"
      href="/"
      onClick={[Function]}
    >
      <img
        className="ui image logo"
        src="magazineluiza.png"
      />
    </a>
    <div
      aria-expanded={false}
      className="ui item dropdown active"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      role="listbox"
      tabIndex={0}
    >
      <div
        aria-live="polite"
        className="text"
        role="alert"
      >
        Comercial
      </div>
      <i
        aria-hidden="true"
        className="dropdown icon"
        onClick={[Function]}
      />
      <div
        className="menu transition"
      >
        <a
          className="item"
          href="/home"
          onClick={[Function]}
          role="option"
        >
          Inicio
        </a>
        <a
          className="item"
          href="/home"
          onClick={[Function]}
          role="option"
        >
          Inicio
        </a>
        <a
          className="item"
          href="/home"
          onClick={[Function]}
          role="option"
        >
          Inicio
        </a>
      </div>
    </div>
    <div
      aria-expanded={false}
      className="ui item dropdown"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      role="listbox"
      tabIndex={0}
    >
      <div
        aria-live="polite"
        className="text"
        role="alert"
      >
        Logística
      </div>
      <i
        aria-hidden="true"
        className="dropdown icon"
        onClick={[Function]}
      />
      <div
        className="menu transition"
      >
        <a
          className="item"
          href="/home"
          onClick={[Function]}
          role="option"
        >
          Inicio
        </a>
        <a
          className="item"
          href="/home"
          onClick={[Function]}
          role="option"
        >
          Inicio
        </a>
        <a
          className="item"
          href="/home"
          onClick={[Function]}
          role="option"
        >
          Inicio
        </a>
      </div>
    </div>
    <div
      aria-expanded={false}
      className="ui item dropdown"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      role="listbox"
      tabIndex={0}
    >
      <div
        aria-live="polite"
        className="text"
        role="alert"
      >
        Tema
      </div>
      <i
        aria-hidden="true"
        className="dropdown icon"
        onClick={[Function]}
      />
      <div
        className="menu transition"
      >
        <a
          className="item"
          href="/theme"
          onClick={[Function]}
          role="option"
        >
          Tema
        </a>
      </div>
    </div>
    <div
      className="right menu logout"
    >
      <div
        aria-expanded={false}
        className="ui item dropdown"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        role="listbox"
        tabIndex={0}
      >
        <div
          aria-live="polite"
          className="text"
          role="alert"
        >
          Olá, CBT_SILVA
        </div>
        <i
          aria-hidden="true"
          className="dropdown icon"
          onClick={[Function]}
        />
        <div
          className="menu transition"
        >
          <a
            className="item"
            href="/logout"
            onClick={[Function]}
            role="option"
          >
            Logout
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
